x-pack/plugins/maps/config.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+export const configSchema = schema.object({
+  enabled: schema.boolean({ defaultValue: true }),
+  showMapVisualizationTypes: schema.boolean({ defaultValue: false }),
+  // flag used in functional testing
+  showMapsInspectorAdapter: schema.boolean({ defaultValue: false }),
+  // flag used in functional testing
+  preserveDrawingBuffer: schema.boolean({ defaultValue: false }),
+  // flag used to enable/disable vector-tiles
+  enableVectorTiles: schema.boolean({ defaultValue: false }),
+});
+
+export type ConfigSchema = TypeOf<typeof configSchema>;
```

Comment:
How about renaming to something like MapsXPackConfig, this is more inline with how some of the other plugins have named there configuration schema, like APM - https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/server/index.ts#L37